File: imperium-common/src/main/java/com/xpdustry/imperium/common/account/Account.java
package com.xpdustry.imperium.common.account;

import java.time.Duration;
import java.time.Instant;
import org.jspecify.annotations.Nullable;

public record Account(
        int id,
        String username,
        Duration playtime,
        Instant creation,
        boolean legacy,
        Rank rank,
        @Nullable Long discord) {}

File: imperium-common/src/main/kotlin/com/xpdustry/imperium/common/account/Account.kt
// SPDX-License-Identifier: GPL-3.0-only
package com.xpdustry.imperium.common.account

import kotlin.time.Duration
import kotlin.time.Instant

data class Account(
    val id: Int,
    val username: String,
    val discord: Long?,
    val games: Int,
    val playtime: Duration,
    val creation: Instant,
    val legacy: Boolean,
    val rank: Rank,
)

data class LegacyAccount(
    val id: Int,
    val games: Int,
    val playtime: Duration,
    val rank: Rank,
    val achievements: Set<Achievement>,
)
